standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -1251,4 +1251,19 @@ public static String getHttpPath(String httpPath) {
     }
     return httpPath;
   }
+
+  /**
+   * Function to get the base name of the column type.
+   * @param colType column type
+   * @return base name of the column type
+   */
+  public static String getBaseNameOfColType(String colType) {
+    // char, varchar types can have parameters such as
+    // char/varchar length. We only need the base name.
+    int idx = colType.indexOf('(');
```

Comment:
Can it be stored with extra whitespace characters, like `varchar (50)`? 

To be honest, I'm a pretty surprised that we don't have existing code to split the type name and the length.

Reply:
The `colType` that you see here is retrieved from metastore DB. I think its an optimisation to not maintain 2 column in metastore DB - one for type and one for length / other information (btw this information is only for char, varchar or other types).

Also similar implementation is present in TypeInfoUtils#getBaseName(String) which does this conversion. But this would require adding `serde` module into `standalone-metastore` which seems to me like an overkill for this fix.

Reply:
Got it, thx.

Reply:
Added trims as well just for safety if we encounter such a situation.